rosetta/retriever/retriever.go

```diff
@@ -350,12 +352,20 @@ func (r *Retriever) operations(txID flow.Identifier, events []flow.Event) ([]*ob
 	// Now we can convert each event to an operation, as they are both filtered for
 	// only supported ones and properly ordered.
 	ops := make([]*object.Operation, 0, len(filtered))
-	for index, event := range filtered {
+
+	// Keep track of event indices while ignoring skipped events.
+	index := 0
+
+	for _, event := range filtered {
 		op, err := r.convert.EventToOperation(uint(index), event)
+		if errors.Is(err, converter.ErrIrrelevant) {
```

Comment:
I'm not sure if this is the right way to do things - `Converter` is an interface described in the `retriever` package, in `rosetta/retriever/converter.go`, and here we're comparing its error to an error from `rosetta/converter` package.

Reply:
The retriever package is the one that should declare the errors it cares about, since it is the consumer of the interface and thus dictates its rules.